service/status-report/main.go

```diff
@@ -138,10 +141,46 @@ func main() {
 	status := providerclient.NewStorageClientStatus().
 		SetPlatformVersion(pltVersion).
 		SetOperatorVersion(oprVersion)
-	if _, err = providerClient.ReportStatus(ctx, storageClient.Status.ConsumerID, status); err != nil {
+	statusResponse, err := providerClient.ReportStatus(ctx, storageClient.Status.ConsumerID, status)
+	if err != nil {
 		klog.Exitf("Failed to report status of storageClient %v: %v", storageClient.Status.ConsumerID, err)
 	}
 
+	versionFromResponse, err := getVersionFromChannel(statusResponse.DesiredClientOperatorChannel)
+	if err != nil {
+		klog.Exitf("Failed to map channel name from status response to version: %v", err)
+	}
+
+	var updateAnnotation bool
+	if storageClient.GetAnnotations()[desiredSubscriptionChannelAnnotationKey] == "" {
```

Comment:
storageClient metadata might not contain an annotation map

```suggestion
	anno := storageClient.GetAnnotations();
	if  anno == nil || anno[desiredSubscriptionChannelAnnotationKey] == "" {
```

Reply:
Removed code which extracts version from channel.